pkg/destroy/azure/azure.go

```diff
@@ -0,0 +1,222 @@
+}
+
+func deletePublicRecords(ctx context.Context, dnsClient dns.ZonesClient, recordsClient dns.RecordSetsClient, logger logrus.FieldLogger, rgName string) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	// collect records from private zones in rgName
+	var errs []error
+	for zonesPage, err := dnsClient.ListByResourceGroup(ctx, rgName, to.Int32Ptr(100)); zonesPage.NotDone(); err = zonesPage.NextWithContext(ctx) {
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "failed to list private zone"))
+			continue
+		}
+		for _, zone := range zonesPage.Values() {
+			if zone.ZoneType == dns.Private {
+				if err := deletePublicRecordsForZone(ctx, dnsClient, recordsClient, logger, rgName, to.String(zone.Name)); err != nil {
+					errs = append(errs, errors.Wrapf(err, "failed to delete public records for %s", to.String(zone.Name)))
+					continue
+				}
+			}
+		}
+	}
+	return utilerrors.NewAggregate(errs)
+}
+
+func deletePublicRecordsForZone(ctx context.Context, dnsClient dns.ZonesClient, recordsClient dns.RecordSetsClient, logger logrus.FieldLogger, zoneGroup, zoneName string) error {
+	// collect all the records from the zoneName
+	allPrivateRecords := sets.NewString()
+	for recordPages, err := recordsClient.ListByDNSZone(ctx, zoneGroup, zoneName, to.Int32Ptr(100), ""); recordPages.NotDone(); err = recordPages.NextWithContext(ctx) {
+		if err != nil {
+			return err
+		}
+		for _, record := range recordPages.Values() {
+			if t := toRecordType(to.String(record.Type)); t == dns.SOA || t == dns.NS {
+				continue
+			}
+			allPrivateRecords.Insert(fmt.Sprintf("%s.%s", to.String(record.Name), zoneName))
+		}
+	}
+
+	sharedZones, err := getSharedDNSZones(ctx, dnsClient, zoneName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find shared zone for %s", zoneName)
+	}
+	for _, sharedZone := range sharedZones {
+		logger.Debugf("removing matching private records from %s", sharedZone.Name)
+		for recordPages, err := recordsClient.ListByDNSZone(ctx, sharedZone.Group, sharedZone.Name, to.Int32Ptr(100), ""); recordPages.NotDone(); err = recordPages.NextWithContext(ctx) {
+			if err != nil {
+				return err
+			}
+			for _, record := range recordPages.Values() {
+				if allPrivateRecords.Has(fmt.Sprintf("%s.%s", to.String(record.Name), sharedZone.Name)) {
+					resp, err := recordsClient.Delete(ctx, sharedZone.Group, sharedZone.Name, to.String(record.Name), toRecordType(to.String(record.Type)), "")
+					if err != nil {
+						if wasNotFound(resp.Response) {
+							logger.WithField("record", to.String(record.Name)).Debug("already deleted")
+							continue
+						}
+						return errors.Wrapf(err, "failed to delete record %s in zone %s", to.String(record.Name), sharedZone.Name)
+					}
+					logger.WithField("record", to.String(record.Name)).Info("deleted")
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// getSharedDNSZone returns the all parent public dns zones for privZoneName in decreasing order of closeness.
```

Comment:
`getSharedDNSZone` -> `getSharedDNSZones`

Reply:
Oops i forgot that :) fixed
https://github.com/openshift/installer/compare/3ef6475d436eee6860c8315a6320dc006c67c663...1bd15ad1005263bf9a9c926eb42c927fca4de33f